designer/client/src/components/toolbars/test/buttons/TestWithFormButton.tsx

```diff
@@ -87,23 +83,20 @@ function TestWithFormButton(props: Props) {
                     ...obj,
                     [uiParam.name]: paramValues[uiParam.name],
                 }),
-                {}
+                {},
             );
             const request: SourceWithParametersTest = {
                 sourceId: selectedSource as string,
                 parameterExpressions: parameters,
             };
             dispatch(testProcessWithParameters(processId, request, processToDisplay));
         },
-        [sourceParameters, selectedSource]
+        [sourceParameters, selectedSource],
     );
 
     useEffect(() => {
-        if (available) dispatch(fetchTestFormParameters(processToDisplay));
-    }, [available, processToDisplay]);
-
-    useEffect(() => {
-        setAvailable(isAvailable);
+        setAvailable(isAvailable());
+        if (isAvailable()) dispatch(fetchTestFormParameters(processToDisplay));
     }, [testCapabilities]);
```

Comment:
`processToDisplay` as dependecy is not needed here?

Reply:
It was causing the problem - `processToDisplay` was changed but `testCapabilities` was not yet fetched (for changed process). `processToDisplay` triggers `testCapabilities` and `testCapabilities` triggers `fetchTestFormParameters`